ui/apps/platform/cypress/integration/accessControl/accessControlAuthProviders.test.js

```diff
@@ -3,51 +3,57 @@ import {
     selectors,
     accessModalSelectors,
 } from '../../constants/AccessControlPage';
-import { permissions as permissionsApi } from '../../constants/apiEndpoints';
+import * as api from '../../constants/apiEndpoints';
 import sampleCert from '../../helpers/sampleCert';
 import { generateNameWithDate, getInputByLabel } from '../../helpers/formHelpers';
+import { visit, visitWithPermissions } from '../../helpers/visit';
 import updateMinimumAccessRoleRequest from '../../fixtures/auth/updateMinimumAccessRole.json';
 
 import withAuth from '../../helpers/basicAuth';
 
-// TODO Fix 'v1/authProviders*' without initial slash and with asterisk in apiEndpoints?
-const authProvidersApi = {
-    list: '/v1/authProviders',
-    create: '/v1/authProviders',
+const h1 = 'Access Control';
+const h2 = 'Auth providers';
+
+const routeMatcherMap = {
+    authProviders: {
+        method: 'GET',
+        url: api.auth.authProviders,
+    },
+    roles: {
+        method: 'GET',
+        url: api.roles.list,
+    },
+    groups: {
+        method: 'GET',
+        url: api.groups.list,
+    },
 };
 
-const mypermissionApi = '/v1/mypermissions';
+function visitAuthProviders(staticResponseMap) {
+    visit(authProvidersUrl, { routeMatcherMap }, staticResponseMap);
 
-const groupsApi = {
-    list: '/v1/groups',
-    batch: '/v1/groupsbatch',
-};
+    cy.get(selectors.breadcrumbNav).should('not.exist');
+    cy.get(`${selectors.h1}:contains("${h1}")`);
+    cy.get(`${selectors.navLinkCurrent}:contains("${h2}")`);
+    cy.get(selectors.list.createButton).should('have.text', 'Create auth provider');
+    cy.contains(selectors.h2, /^\d+ results? found$/).should('exist');
+}
 
-const h1 = 'Access Control';
-const h2 = 'Auth providers';
+function visitAuthProviderById(id, staticResponseMap) {
+    visit(`${authProvidersUrl}/${id}`, { routeMatcherMap }, staticResponseMap);
+}
 
 describe('Access Control Auth providers', () => {
     withAuth();
 
-    function visitAuthProviders(saveProviderMock = {}) {
-        cy.intercept('GET', authProvidersApi.list).as('GetAuthProviders');
-        cy.intercept('GET', mypermissionApi).as('GetMyPermissions');
-        cy.intercept('POST', authProvidersApi.create, saveProviderMock).as('CreateAuthProvider');
-        cy.visit(authProvidersUrl);
-        cy.wait('@GetAuthProviders');
-        cy.wait('@GetMyPermissions');
-    }
-
     it('displays alert if no permission', () => {
-        cy.intercept('GET', permissionsApi.mypermissions, {
+        const permissionsStaticResponse = {
             fixture: 'auth/mypermissionsMinimalAccess.json',
-        }).as('GetMyPermissions');
-        cy.visit(authProvidersUrl);
-        cy.wait('@GetMyPermissions');
+        };
+        visitWithPermissions(authProvidersUrl, permissionsStaticResponse);
```

Comment:
`visitWithPermissions()` takes 4 arguments, but we're only passing 2 parameters here?

Reply:
Good eyes. Yes, the others are optional so tests can call same function:
* mock permissions only
* mock permissions and also wait on requests
* mock permissions and also wait on requests and also mock responses

I added JSDoc `param` types in case people recognize name in square brackets as notation for optional:
* `[requestConfig]`
* `[staticResponseMap]`